caffeine@patapon.info/extension.js

```diff
@@ -191,67 +395,251 @@ class Caffeine extends PanelMenu.Button {
         }
         return inFullscreen;
     }
-
+    
     toggleFullscreen() {
+        this._manageScreenBlankState(false);
         Mainloop.timeout_add_seconds(2, () => {
-            if (this.inFullscreen && !this._apps.includes('fullscreen')) {
+            if (this.inFullscreen && !this._appInhibitedData.has('fullscreen')) {
                 this.addInhibit('fullscreen');
-                this._manageNightLight('disabled');
+                this._manageNightLight(false, false);
             }
         });
 
-        if (!this.inFullscreen && this._apps.includes('fullscreen')) {
+        if (!this.inFullscreen && this._appInhibitedData.has('fullscreen')) {
             this.removeInhibit('fullscreen');
-            this._manageNightLight('enabled');
+            this._manageNightLight(true, false);
         }
     }
 
     toggleState() {
+        this._manageScreenBlankState(false);
         if (this._state) {
-            this._apps.forEach(appId => this.removeInhibit(appId));
-            this._manageNightLight('enabled');
-        } else {
+            this._removeTimer(false);
+            this._appInhibitedData.forEach((data, appId) => 
+                this.removeInhibit(appId)
+            );
+            this._manageNightLight(true, false);            
+        } else {     
             this.addInhibit('user');
-            this._manageNightLight('disabled');
-        }
+            this._manageNightLight(false, false);
+        }        
```

Comment:
There's quite a lot of trailing spaces in the file, they should be removed

Reply:
Most of the time, I try to get rid of it. I'll correct this.